bundles/org.openhab.binding.knx/src/test/java/org/openhab/binding/knx/internal/itests/Back2BackTest.java

```diff
@@ -332,22 +332,39 @@ void testDpt3() {
 
     @Test
     void testDpt5() {
-        // TODO add tests for more subtypes
+        helper("5.001", new byte[] { 0 }, new QuantityType<>("0 %"));
+        helper("5.001", new byte[] { (byte) 0xff }, new QuantityType<>("100 %"));
+        // fallback: PercentType
         helper("5.001", new byte[] { 0 }, new PercentType(0));
         helper("5.001", new byte[] { (byte) 0x80 }, new PercentType(50));
         helper("5.001", new byte[] { (byte) 0xff }, new PercentType(100));
 
+        helper("5.003", new byte[] { 0 }, new QuantityType<>("0 °"));
+        helper("5.003", new byte[] { (byte) 0xff }, new QuantityType<>("360 °"));
+        helper("5.004", new byte[] { 0 }, new QuantityType<>("0 %"));
+        helper("5.004", new byte[] { (byte) 0x64 }, new QuantityType<>("100 %"));
+        helper("5.004", new byte[] { (byte) 0xff }, new QuantityType<>("255 %"));
+        // PercentType cannot encode values >100%, not supported for 5.004
```

Comment:
Out of curiosity, how does this work as it can't encode 100+, isn't FF expected to be 100% ? or a special value like 'turn on with previous value'

Reply:
This data type is defined in the KNX spec as 0..255%, i.e. 0xff wire representation should really be 255%.
Since we introduced UOM, KNX binding maps DPT 5.004 to QuantityType, as it can represent the full range of values.

> Attention: With the introduction of Unit of Measurement (UoM) support, some data types have changed (see number channel below):
> Data type for DPT 5.004 (Percent 8bit, 0 -> 255%) has changed from PercentType to QuantityType.

Converting from QuantityType to PercentType will cause an exception if the value is >100, as the ctor of PercentType checks the input. I have not implemented a special handling to allow a better interaction with PercentType, as I think this DPT is quite uncommon.

Reply:
Thanks for taking the time.